include/real/exact_number.hpp

```diff
@@ -264,6 +272,116 @@ namespace boost {
+
+            	int a_size = this->digits.size();
+            	int b_size = other.digits.size();
+
+                const int max_length = std::max(a_size, b_size);
+
+                if (max_length <= KARATSUBA_THRESHOLD) {
+                    this->standard_multiplication(other, base);
+                    return;
+                }
+
+
+                // appending zeroes in front to make sizes of a & b equal
+                std::vector<T> final_a, final_b;
+                if (a_size < max_length) {
+                	std::vector<T> zeroes(max_length - a_size, 0);
+                	final_a.reserve(max_length);
+                	final_a.insert(final_a.end(), zeroes.begin(), zeroes.end());
+                	final_a.insert(final_a.end(), this->digits.begin(), this->digits.end());
+                	final_b = other.digits;
+                } else if (b_size < max_length) {
+                	std::vector<T> zeroes(max_length - b_size, 0);
+                	final_b.reserve(max_length);
+                	final_b.insert(final_b.end(), zeroes.begin(), zeroes.end());
+                	final_b.insert(final_b.end(), other.digits.begin(), other.digits.end());
+                	final_a = this->digits;
+                } else {
+                	final_a = this->digits;
+                	final_b = other.digits;
+                }
+
+                const int left_half_length = max_length / 2;
+                const int right_half_length = max_length - left_half_length;
+
+
+                //   a = al * base^(left_half_length) + ar
+                std::vector<T> al (final_a.begin(), final_a.begin() + left_half_length);
+                std::vector<T> ar (final_a.begin() + left_half_length, final_a.end());
+
+                //   b = al * base^(left_half_length) + br
+                std::vector<T> bl (final_b.begin(), final_b.begin() + left_half_length);
+                std::vector<T> br (final_b.begin() + left_half_length, final_b.end()); 
+
+                exact_number<T> exact_al (al, true);
+                exact_number<T> exact_ar (ar, true);
+                exact_number<T> exact_bl (bl, true);
+                exact_number<T> exact_br (br, true);
+
+                exact_al.normalize();
+                exact_bl.normalize();
+                exact_ar.normalize();
+                exact_br.normalize();
+
+                exact_number<T> sum_al_ar (al, true);
+                exact_number<T> sum_bl_br (bl, true);
+
+                // sum_al_ar = al + ar
+                // sum_bl_br = bl + br
+                sum_al_ar.add_vector(exact_ar, base - 1);
+                sum_bl_br.add_vector(exact_br, base - 1);
+
+
+                exact_al.karatsuba_multiplication(exact_bl, base);
+                sum_al_ar.karatsuba_multiplication(sum_bl_br, base);
+                exact_ar.karatsuba_multiplication(exact_br, base);
+
+                // exact_al = al * bl
+                // sum_al_ar = (al + ar) * (bl + br)
+                // exact_ar = ar * br
```

Comment:
remove deprecated code.

Reply:
actually it is an explanation of what these variables are.

Reply:
haha ok ok, I just saw code in a coment and add the comment. When that is the case it is best to use /* */ comments as they are more standard for documentation code and not to comment deprecated code. Also put a little title.

/*
    Variables explanation
    exact_al = al * bl
    sum_al_ar = (al + ar) * (bl + br)
    exact_ar = ar * br
*/

Reply:
Ok, will do it.